src/config_dir_watcher.py

```diff
@@ -58,12 +59,15 @@ def start_watchdog(self):
         if "JUJU_CONTEXT_ID" in new_env:
             new_env.pop("JUJU_CONTEXT_ID")
 
+        juju_bin = (
+            "/usr/bin/juju-exec" if Path("/usr/bin/juju-exec").exists() else "/usr/bin/juju-run"
```

Comment:
Not sure I understand the question. If `juju-exec` exists we use it, if not we use `juju-run`.

Or maybe you meant what version of juju will provide each binary? In that case, I am not sure where the cutoff is.

Reply:
Sorry the question wasn't clear. What makes it so that this path to exists? You sort of answered it by saying that it's determined by the juju version

Reply:
Yes, that is managed by Juju with the charm base images. In recent version, `juju-exec` was removed in favor of `juju-run`.